webpack/InsightsCloudSync/Components/NoTokenEmptyState.js

```diff
@@ -0,0 +1,86 @@
+import React, { useState } from 'react';
+import { useDispatch } from 'react-redux';
+import {
+  Button,
+  TextInput,
+  FormGroup,
+  GridItem,
+  Grid,
+  EmptyState,
+  EmptyStateIcon,
+  EmptyStateVariant,
+  EmptyStateBody,
+} from '@patternfly/react-core';
+import PropTypes from 'prop-types';
+import { ExternalLinkAltIcon, RocketIcon } from '@patternfly/react-icons';
+import { put } from 'foremanReact/redux/API';
+import { translate as __ } from 'foremanReact/common/I18n';
+import { foremanUrl } from '../../ForemanRhCloudHelpers';
+import { SETTINGS_API_KEY } from '../InsightsCloudSyncConstants';
+
+export const NoTokenEmptyState = ({ reloadData }) => {
+  const [token, setToken] = useState('');
+  const dispatch = useDispatch();
+  const settingsUrl = foremanUrl('/api/v2/settings/rh_cloud_token/');
+  const onSave = () => {
+    dispatch(
+      put({
+        key: SETTINGS_API_KEY,
+        url: settingsUrl,
+        params: { setting: { value: token } },
+        handleSuccess: reloadData,
+      })
+    );
+  };
```

Comment:
nice work! though Ithink it should be a `POST` to `hits_controller` which will save the setting and trigger the sync task.

Reply:
Added a path and a function for that

Reply:
Totally of topic, but when I've seen this, I wondered: would there be an interest (and need) for having a "real" api client in JS, that can build those paths etc automatically? Like we have for Ruby and Python?

Reply:
> Totally of topic, but when I've seen this, I wondered: would there be an interest (and need) for having a "real" api client in JS, that can build those paths etc automatically? Like we have for Ruby and Python?

I think you're talking about something like this one: https://github.com/railsware/js-routes.
Are you sure you want the packaging hassle of it?